# SOAL ETS 2019
by: Sumihar Christian 

In [18]:
from math import sqrt
# define objective function 
def obj(x): 
    return sqrt((x[0]-20)**2+(x[1]-0)**2) + sqrt((x[0]-0)**2 + (x[1]-20)**2)    

In [19]:
#define constraint
def cons3 (x): 
    return (x[0])**2 + (x[1])**2 - 400**2

In [20]:
# define type of constraint, with boundary
cons = {
    'type':'ineq', 
    'fun':cons3, 
    'bounds':[0,0]
}

In [21]:
# give initial condition, and solve the equation using KKT from scipy.minimize
from scipy.optimize import minimize
x0 = (0,0)
solution = minimize(obj, x0, constraints=cons)

In [22]:
solution

     fun: 772.2339042583401
     jac: array([1.41136169, 1.41516113])
 message: 'Optimization terminated successfully.'
    nfev: 86
     nit: 22
    njev: 21
  status: 0
 success: True
       x: array([282.4747214 , 283.21022543])

## from solution we get 
$$x = 282.474 \\
y = 283.210 \\
$$

where the sum between distance of $s_2$ to A, and distance of $s_2$ to B is 772.23309